scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/service/ScheduleManagerTest.java

```diff
@@ -838,20 +847,29 @@ public void testSynchronizeSchedules_with_both_policy_and_schedules_existed_and_
 		verify(scheduleJobManager, never()).createSimpleJob(any());
 		verify(scheduleJobManager, never()).deleteJob(anyString(), anyLong(), any());
 	}
-
+	@Captor ArgumentCaptor<RecurringScheduleEntity> recurringCaptor;
+	@Captor ArgumentCaptor<SpecificDateScheduleEntity> specificDateCaptor;
 	private void assertCreateSchedules(Schedules schedules, SpecificDateScheduleEntity specificDateScheduleEntity,
 			RecurringScheduleEntity recurringScheduleEntity, int noOfSpecificDateSchedules,
 			int noOfDOMRecurringSchedules, int noOfDOWRecurringSchedules) {
-
+		
 		if (schedules.getSpecificDate() != null) {
+			Mockito.verify(specificDateScheduleDao, Mockito.times(noOfSpecificDateSchedules)).create(specificDateCaptor.capture());
+			List<SpecificDateScheduleEntity> specificDateList = specificDateCaptor.getAllValues();
+			int i = 0;
 			for (SpecificDateScheduleEntity foundSpecificDateScheduleEntity : schedules.getSpecificDate()) {
-				Mockito.verify(specificDateScheduleDao, Mockito.times(1)).create(foundSpecificDateScheduleEntity);
+				assertEquals(foundSpecificDateScheduleEntity,specificDateList.get(i));
+				i++;
```

Comment:
Just for information, the assert could have used list contains, as we cannot assume list to be sorted.